packages/sdk/src/sdkBase.ts

```diff
@@ -1019,75 +1030,40 @@ export class NxtpSdkBase {
     requestContext: RequestContext,
     methodContext: MethodContext,
   ): Promise<BigNumber> {
-    this.logger.info("Calculate gas in token for prepare", requestContext, methodContext, {
+    this.logger.info("Calculating gas fee in token for prepare", requestContext, methodContext, {
       chainId,
       assetId,
       decimals,
     });
 
-    const { provider } = this.config.chainConfig[chainId] ?? {};
-
-    const priceOracleContract = getDeployedPriceOracleContract(chainId);
     const gasLimitForPrepare = BigNumber.from(GAS_ESTIMATES.prepare);
     let totalCost = constants.Zero;
-    if (priceOracleContract && priceOracleContract.address && provider) {
-      const priceOracleAddress = priceOracleContract.address;
-      const ethPriceInUsd = await getTokenPrice(priceOracleAddress, constants.AddressZero, provider);
-      const tokenPriceInUsd = await getTokenPrice(priceOracleAddress, assetId, provider);
-      const gasPrice = await this.getGasPrice(chainId);
+    try {
+      const ethPriceInUsd = await this.chainReader.getTokenPrice(chainId, constants.AddressZero);
+      const tokenPriceInUsd = await this.chainReader.getTokenPrice(chainId, assetId);
+      const gasPrice = await this.getGasPrice(chainId, requestContext);
       const gasAmountInUsd = gasPrice.mul(gasLimitForPrepare).mul(ethPriceInUsd);
       const tokenAmountForGasFee = tokenPriceInUsd.isZero()
         ? constants.Zero
         : gasAmountInUsd.div(tokenPriceInUsd).div(BigNumber.from(10).pow(18 - decimals));
       totalCost = totalCost.add(tokenAmountForGasFee);
+    } catch (e) {
+      this.logger.error(
+        "Error estimating gas fee in token for prepare",
+        requestContext,
+        methodContext,
+        jsonifyError(e),
+        {
+          chainId,
+          assetId,
+          decimals,
+        },
+      );
     }
 
     return totalCost;
   }
 
-  /**
-   * Estimates hardcoded fee for fulfill transactions
-   *
-   * @param chainId - The network indentifier
-   * @param assetId - The asset address
-   * @param decimals - The asset decimals
-   * @returns Gas fee for fulfill in token
-   */
-  async estimateHardcodedFeeForFulfill(
```

Comment:
I removed this entirely; it only seemed to be used within `estimateFeeForMetaTx` and now we can just call the chain reader method directly for this.

Please correct me if I missed something! thanks